### PR TITLE
chore(guidelines): Add contributing guidelines

### LINKED ISSUE #2: Update Readme for Contributing Guidelines
Needs to cover:

- should tell user how to setup local dev environment
- should give list of helpful commands for tests/linting before pushing/PR
- guide/encourage users to setup and follow git-flow for development
- add general coding guidelines
- add contributing doc for copyright ownership clause

### PR DESCRIPTION
**Pull request by jainmickey**
fixes #2

### CODE REVIEW

File: README.dev.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        verbose_name_plural = _(u'some classes')
+        ordering = ['name']
+
+    def __unicode__(self):
+        return self.name
+```
+
+# Setup
+
+Minimum requirements: **python3, pip & postgres**.
```
Comment (akarambir) [inline README.dev.md]: We can mention in next line that:

*We advise to use `virtualenv` with python3 interpretor.*

File: README.dev.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+## Getting up and running
+
+```
+pip install -r requirements.txt
+createdb askcoding
+python manage.py migrate
+```
+
+[install-postgres]: https://wiki.postgresql.org/wiki/Detailed_installation_guide
```
Comment (akarambir) [inline README.dev.md]: All the links like these can go to the bottom for clarity. Look at readme for example.

File: README.dev.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+# Git Flow
+
+Vincent Driessen's branching model is a git branching and release management strategy that helps developers keep track of features, hotfixes and releases in projects. This workflow has lot of commands to type and remember, though, so there's also the git-flow library of git subcommands that helps automate some parts of the flow to make working with it a lot easier. To learn more about git-flow look at this [link][gitflow-model]
```
Comment (akarambir) [inline README.dev.md]: We can just say we follow git-flow for development/release. No need to explain gitflow. A link would suffice.

File: README.dev.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+* Testing
+```
+py.test --cov -v --pdb
+```
+
+* Linting
+```
+flake8 .
+```
```
Comment (akarambir) [inline README.dev.md]: We can also mention that: We *try* to keep code coverage more than 90%. Check coveralls for this.

File: README.md [hunk 5]
```diff
@@ -47,6 +47,10 @@ AskCoding is a Django/Python based website. We aim to use these technologies:
 - EditorConfig/isort for formatting Python files.
 - Py.test for testing.
 
+## Installing the AskCoding Development environment
```
Comment (akarambir) [inline README.md]: Please move this below *Contribution` heading. So users first read contribution and then about setting dev env

File: README.dev.md (hunk 1 above)
Comment (jainmickey) [inline README.dev.md]: done

File: README.dev.md (hunk 2 above)
Comment (jainmickey) [inline README.dev.md]: done

File: README.dev.md (hunk 3 above)
Comment (jainmickey) [inline README.dev.md]: done

File: README.dev.md (hunk 4 above)
Comment (jainmickey) [inline README.dev.md]: done

File: README.md (hunk 5 above)
Comment (jainmickey) [inline README.md]: done

**Pull request merged**